packages/intent-filler/src/strategies/basic.ts

```diff
@@ -369,8 +382,33 @@ export class BasicFiller implements FillerStrategy {
 	/**
 	 * Estimates gas for handling POST requests in the source chain
 	 */
-	private async estimateGasForPost(order: Order): Promise<number> {
-		// TODO: Implement this
+	private async estimateGasForPost(
+		order: Order,
+		providers: { sourceProvider: ethers.providers.Provider; destProvider: ethers.providers.Provider },
+	): Promise<number> {
+		const postRequest: IPostRequest = {
+			source: order.destChain,
+			dest: order.sourceChain,
+			body: this.constructRedeemEscrowRequestBody(order, providers.sourceProvider),
+			timeoutTimestamp: 0n,
+			nonce: await this.getHostNonce(providers.destProvider, order.destChain),
+			from: addresses.IntentGateway[order.destChain as keyof typeof addresses.IntentGateway]!,
+			to: addresses.IntentGateway[order.sourceChain as keyof typeof addresses.IntentGateway]!,
+		}
+
+		const { root, proof } = generateRootWithProof(postRequest)
+		const latestStateMachineHeight = await this.getHostLatestStateMachineHeight(
+			providers.sourceProvider,
+			order.destChain,
```

Comment:
On EVM chains, the hyperbridge is the only state machine, so you should fetch hyperbridge's latest state machine height.

Reply:
Yes

Reply:
Done